x-pack/plugins/apm/public/hooks/useAnomalyDetectionSettings.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useFetcher } from './useFetcher';
+
+export function useAnomalyDetectionSettings() {
+  return useFetcher(
+    (callApmApi) =>
+      callApmApi({
+        pathname: `/api/apm/settings/anomaly-detection`,
+      }),
+    [],
+    { showToastOnError: false }
```

Comment:
Do we handle potential http errors elsewhere in the ui?

Reply:
This endpoint throws if the license is not appropriate, and there probably are some other situations as well. There might be other errors too, perhaps from the ML client. In that case, we just don't show the callout because we assume the user cannot set up ML anyway. The errors will be logged server-side.